src/rosdep2/sources_list.py

```diff
@@ -320,9 +321,12 @@ def download_default_sources_list(url=DEFAULT_SOURCES_LIST_URL):
     data = f.read().decode()
     f.close()
     if not data:
-        raise RuntimeError("cannot download defaults file: empty contents")
+        raise DownloadFailure("cannot download defaults file: empty contents")
     # parse just for validation
-    parse_sources_data(data)
+    try:
+        parse_sources_data(data)
+    except InvalidData as e:
+        raise DownloadFailure("Failed to download valid rosdep sources from %s Contents were:{{{%s}}} Parsing error: %s" % (url, data, e))
```

Comment:
> I kind of agree here, since we could conceivably accidentally make malformed sources and then this error would be incorrectly pointing people to problems with their network rather than pointing them at us.

If we are worried about malformed upstream sources, we should add CI in the upstream repo. If it is broken for everyone, people will find the right place to report it even when the error message says its a download failure. I'm +1 on @tfoote's change as is. Or maybe we could rephrase slightly like: `"Failed to download and validate rosdep sources from..."`